Header UIの作成

london-newyork opened the pull request:
# 該当チケット

#43 

# やったこと
- Header UIの作成
- アイコンを仮で使うので、Tailwind CSS と親和性の高い[Heroicons](https://heroicons.com/)の導入
- Tailwind CSSでスタイリングするための必要なファイルなど作成し環境構築

[やっていないこと]
- UI作成効率化のためHeadless UIなど導入しようと思ったが、yarnだとうまくいかず、npmのほうがサポートしていそうだった。一旦今は保留。

# 動作確認

ローカル、ブラウザでの動き確認のみ

# 参考にした記事

とくになし

london-newyork committed:
TailwindCSSを効かせるために新たに必要なものを追加した。

london-newyork committed:
アイコンを入れたいのでTailwindと親和性高いheroicons導入

london-newyork committed:
heroicons導入続き

london-newyork committed:
headerの要素の大部分をスタイリングし終わった。

london-newyork committed:
共通コンポーネントとして移動

london-newyork committed:
リファクタ

tatsuro1997 commented on components/Header.tsx:
```diff
@@ -1,25 +1,71 @@
+      </ul> */}
+      <header className='header-wrapper'>
+        <div className='header-contents'>
+          <div>
```
Buttonコンポーネント内で`<div>`でラップしているため、現状不要なように見えます。
Buttonコンポーネント内の`<div>`も現状は不要かもしれません。

london-newyork replied:
ありがとうございます。
a1c6bd0f36bb8663769a36401f5bdf27ae6f98e6
で対応しました。

tatsuro1997 commented on components/commons/SVG.tsx:
```diff
@@ -0,0 +1,24 @@
+        stroke={stroke}
+        className={className}
+      >
+        <path stroke-linecap='round' stroke-linejoin='round' d={d} />
```
コンソールでエラーになっているので確認お願いします。

london-newyork replied:
[こちら](https://github.com/zero2hero-jp/secbokapp-front/pull/46/commits/df201ffaf360a8494d53cc1351b8645356d1a889#:~:text=0%2024%2024%27-,stroke,-%2Dwidth%3D%7BstrokeWidth%7D)で対応しました。
df201ffaf360a8494d53cc1351b8645356d1a889

tatsuro1997 reviewed (changes requested):
HeaderやFooterなどはuiフォルダ配下にしていただくと良いかと思います。
そうするとimportのパスが長くなると思うので、tsconfigのpathsでエイリアスを指定していただけるとスッキリすると思います。

london-newyork committed:
ディレクトリ構成を変更
- 完全に好みだがpagesやcomponentsをsrc配下に設定
- こうすることでソースコードであると明示的に示すとともに設定のためのファイルたちと一線を画すと考えています

london-newyork committed:
UIなのでui配下に移動し、import を短く記述するためにtsconfig にpathsを使用

london-newyork committed:
余計なdivタグやfragmentが入ってしまっていたので取り除いた

london-newyork committed:
Vercelのデプロイに失敗したのはbuttonタグ自体をラップしなかったせいかもしれないため、
一旦fragmentでラップ

london-newyork committed:
Vercelデプロイ失敗がbaseUrlの設定かもしれないため修正

london-newyork committed:
デプロイ成功し、以前の失敗の原因はこれではなかったのでラップをはずした

london-newyork commented:
> HeaderやFooterなどはuiフォルダ配下にしていただくと良いかと思います。 そうするとimportのパスが長くなると思うので、tsconfigのpathsでエイリアスを指定していただけるとスッキリすると思います。

ui配下にHeaderやFooter設置しました。
pathsも対応済みです。
3d95ead377fed58f46e7edf3060e03995ed7185d
ありがとうございます🙏

tatsuro1997 commented:
@london-newyork 
Layoutコンポーネントもuiですね！

london-newyork committed:
Layoutもuiなのでui配下に設置

london-newyork committed:
pathsにimport時の短縮のパスを設定続き

london-newyork commented:
> @london-newyork Layoutコンポーネントもuiですね！

ありがとうございます。
15b3f482352fc228c5784e48a63adfa1eb559672
で対応いたしました。

マージの承認いただいているのでマージします。
